Investigation clean

kdaveed opened the pull request:
Two custom entry forms are working right now. One for the skull, one for study design execution. For the  study design execution it is necessary to load the files from the FrSexEst repository. The most important new feature that the existing instance selector is a popup window which makes the page more useable.
The forms show the existing data, but it is not yet possible to modify the added data.

Commit:
Triple definition is now correct.

Commit:
DependencyCalculator creates the VariableDependency
instances. Based on this instances client JS code can
ask the server form different form data.

Commit:
DependencyCalculator takes into account if the dataKey
is different from variableName in the SubformAdder formelement.

Commit:
Java representation of the form models. The getJSON()
method prepares the appropriate form descriptor JS
object for the client.

Commit:
Config corrections. JSON library can now display JSONObject
in a readable way. ArrayLib extension.

Commit:
New descriptor class types. They do not have
functionality.

Commit:
Graph.java can return the dataDependecies
in form of JSONObject. FormConfigurationLoader
is an AJAX servlet called by the form
initialisation by JS.

Commit:
FormDataLoader now works based on the
VariableDependencies of the Graph object.
The hard coded queries were deleted.

Commit:
FormConfigLoader url mapping

Commit:
Tiny Java bug fixes and modifications.

Commit:
obo-fma prefix is set

Commit:
CustomForm is added as field to
EditConfigurationVTwo object.

Commit:
PopUpController has new functionalities.
GenericForm have now the popup window container.
Scrolling is elegantly disabled.

Commit:
Static descriptor JSON objects are
deleted. Everything comes from
the server dynamically
FormLoader loads the config data
by the pageLoad, and just after
displays the form.
DataController waits with the
loading msg at least a half second.

Commit:
menu.n3 is in the third tier

Commit:
listViewConfig.n3 renamed to customListViewConfig.n3

Commit:
FauxPropertyDefinition.n3 were renamed to studyDesignExecutionFauxProperty.n3

Commit:
Events and Organisations were removed from the main menu

Commit:
renamed .n3 config files

Commit:
New menu items for skeletal inventories and investigations.
Two new property groups for skull and vertebral column.
Faux properties for skull, skeletal regions of skull and
single bones of skull.

Commit:
FauxProperty.java and FauxPropertyDaoJena are in the third tier.

Commit:
lib-properties.ftl is in the third tier.

Commit:
GraphProcessor bug fixed. TripleLib now contains the
data and form definitions for skeletal inventories.

Commit:
FormConfiguration class incorporates the Graph
and the Form definition. The client connector java
classes have been modified accordingly.

Commit:
Triple initialisation is simplified with Object-s in the constructor.

Commit:
VariableDependency add the label triple itself.
New SPARQLDataGetter subclass.

Commit:
Util classes.

Commit:
EntryForm generator classes for skeletal inventories.

Commit:
SDE EntryForm definition were simplified.

Commit:
SubFormAdder bug fix.

Commit:
Utils class update

Commit:
RangeUri is considered in the genericForm.ftl.

Commit:
DataController now can calculate formdata from the parentforms. FormLoader sets the rangeuri.

Commit:
Renamed gif

Commit:
FauxProperties are reffering to the newly created objectproperties.

Commit:
TripleLib restructured. Skeletal inventory config
bug fixed.

Commit:
Removed comments

Commit:
Now the GraphProcessor can handle circular input graphs.
With the fromSubject field of the Triple object
we can control the path of the graph discovery.

Commit:
New style classes

Commit:
PopUpController now can insert waiting gif into
divs.

Commit:
TextButtons click event works even after show
and hide actions on the div.

Commit:
Form loading a bit redesigned. The Adder and
the MainForm initializes now the display of
the formelements, not the Form.

Commit:
Patching the branch in the filegraph submodule.

Commit:
DataController does not operate with the PopUps and
waiting gifs any more. This is done by the
Form and FormElement objects.

Commit:
Waiter gif loading is cleared. ImgUi is called
by the PopUpController. html can handle image
sizes as well.

Commit:
Deleted unnecessary css and js files.
New style classes and new css files.
Calling them from FTL files were cleared.

Commit:
GraphLib bug fixed. Input instances are not added
to the inputClasses array any more. UnionForm and
SubGraphInfo alogrithms were simplified. Graph
can now debug dependencies.

Commit:
GraphProcessor is not needed any more. Graph overtakes
it task, and solves it in a more elaborate way.

Commit:
Graph.java gets new fields - graphMap, initialGraphMap -
for representing the graph structure and
calculates it by itself without the GraphProcessor.
The new feature in the graph representation is
that the subgraph keys are not any more nodes but
predicates -inputPredicate-. In this way the arriving
data structure clearer. The Form.java generates the form
descriptor JSON object according to this change.

Commit:
Graph debug routines are moved from Graph.java to DebugLib.java.

Commit:
Bug fixes by graph structure processesing and
triple generation Graph.java.

Commit:
SPARQLQueryDataGetter input parameter substitution
fixed.

Commit:
Library improvements.

Commit:
Log removes, tiny corrections.

Commit:
The form descriptor dataset now contains the predicates
as keys to subforms.

Commit:
Java Datagenerator signes if the triple creation fails.

Commit:
Formloader updated. Handles failed triples creation.
Adds custom style to the mainform title. Some function
rearrangements.

Commit:
Interface related modification. Moreover the FormElements
handles the predicates as datakeys. Selector initializes
its value correctly.

Commit:
JS library updates

Commit:
JS library updates 2

Commit:
FTL updates

Commit:
Removed unused imports.

Commit:
DebugLib space is used more efficiently.

Commit:
mainInputNodes and mainInputValues fields are created in Graph.java.
The graphLib sets the mainInputNodes by the graph discovery. By the
initialization the mainInputValues are set from the WebappConnector.

Commit:
StringUtil.java created. DebugLib.java has now auxiliary functions -
logString(), logTripleList(), logList(), logDataGetter() - in order to
to make the debugging of the Graph.java clearer.

Commit:
Bug fix. InputInstance have overwritten MainInputInstance.

Commit:
Loading the label of the existing instance is now working.

Commit:
subject and object were replaced everywhere to subjectUri and
objectUri.

Commit:
In the three main JS controller some correction were done.
Now the loading of the existing data works properly.

Commit:
Bug fixed by the loading of the existing form data. ready()
function is now clearer.

Commit:
Now FormElements.js checks that a type from the selector cannot
be added twice.

Commit:
Triple configuration corrections for study design execution.

Commit:
Graph.java and VariableDependency.java were moved to graphData package.

Commit:
The existing instance adder js routine did not respond to the
parent, so it blocked the display of the existing data. Now it is fixed.

Commit:
Tiny corrections.

Commit:
Importing Graph.java and VariableDependency.java is now correct.

Commit:
New library functions.

Commit:
RDF application config files. filegraph submodule path refresh.

Commit:
Tiny js modifications.

Commit:
The form data loading routine is now overtaken by the Graph.java from the
FormDataLoader.java. Now the handling of the 'Add all' events is solved more
elegantly with only one AJAX call.

Commit:
New auxiliary functions.

Commit:
DataController generates the option map from array.

Commit:
The 'Add all' routine works properly with only one AJAX
call.

Commit:
QualifiedRestrictionTriple class is added. It creates
the string for the query with the parameter owl:onClass.
TripleLib for study design execution is updated.

Commit:
Add all button appears by the subformadder only if the
there are more then one options.

Commit:
FormGraph is the new class for the definition of the
data structure required by the selection of existing
instances.

Commit:
ImageCell, TableCell and Table are the classes for the
table definition, where the existing intances are
displayed.

Commit:
TripleLib is extended with the FormGraph definition
for the study design execution.

Commit:
FormGraph is added to the FormConfiguration class. Form
now gets the MainGraph instance so that it has access
to the WebappConnector. Moreover now the FormGraph
can retrieve data for the table based on the data
definition.

Commit:
Small corrections, imports. New lib methods.

Commit:
AJAXServlet is the global servlet handling all AJAX request coming from the
client. Now the functionality of the FormConfigurationLoader is moved to
the AJAXServlet. Accordingly the servlet mappings are changed in the
web.xml.

Commit:
DataController has now an AJAX auxiliary function that facilitates the
communication on the client. FormLoader already utilizes this library.

Commit:
ExistingInstanceSelector field is added. Right now the functionality is
not implemented.

Commit:
AJAXError and AJAXController.java are for the request handling.

Commit:
Table and TableCell classes now can return their descriptor JSON objects.

Commit:
New constructors for GraphLib and FormConfiguration to be able to handle
the FormGraph data maps for the existing instances.

Commit:
Small modifications, function renamings.

Commit:
New util functions in VIVO

Commit:
The changes regard only the identation of the code

Commit:
ExistingInstanceSelector class now return its JSON descriptor as well.

Commit:
Study Design Execution formconfiguration is extended with the existing instance
selector table configuration coming from the TripleLib.

Commit:
FormElements.js were restructured to object oriented style. The class
FormElement is the parent of the Selector, Adder and ExistingInstanceSelector
classes. Each subclass implements the initUI() and initData() functions
individually. The order of the parameters were a bit changed as well.
The FormDataLoader were just adopted to the parameter order change.

Commit:
There are cases where multiple type of restriction property has to be
considered. Now the RestrictionTriple.java is prepared for such cases.
In the TripleLib by the study design execution a restriction is called both
with allValuesFrom and someValuesFrom property.
New function in the StringUtil.java.

Commit:
New FormElement - ExistingSelector

Commit:
The mainGraph of the FormGraph instance are set in the FormConfiguration
class. And by the initialisation the FormGraph passes the mainGraph to its
subgraphs.

Commit:
Bug fixed in AJAXController. Not the reques but the response had to be set.

Commit:
For the boneSegment formgraph the triple configuration were updated.

Commit:
'subGraph.dataTriples.add(triple)' line was at wrong place.
Now the image loader subform contains the right triples.

Commit:
Config data retrival corrections in java.

Commit:
Identation setting

Commit:
Two types of cells for showing information about selectable existing
instances.

Commit:
TableRow displays the cells based on data and the configuration.

Commit:
InstanceSelector handles the popup window for the existing instance
selection. It is called ExistingInstanceSelector object int the
Formelements.js.

Commit:
script import updates in ftl files

Commit:
New util function in DataController.js

Commit:
TextButton now considers the style coming as parameter as well. The click
event by Button instances works now if the button is hidden and showed
dynamically.

Commit:
Small corrections.

Commit:
Identation editing

Commit:
JS util function and one css class correction

Commit:
PopUpController can now handle custom divs as parameter.

Commit:
Tiny corrections

Commit:
ExistingInstanceSelector adds the data selected by the InstanceSelector
object properly.

Commit:
FormLoader now adds the cancel button, and by the editing the
submit button is hidden. So the form by the edit works actually
just as a data view.

Commit:
InstanceSelector prepares the selected and selectable instances properly
and creates the two table with the TableRow instances.

Commit:
filegraph submodule path

Commit:
CSS and JS import is solved more clearly with FTL macros.

Commit:
ExistingFormDataLoader class were deleted (consequently the url mapping as well).
The AJAXController overtakes its task.

Commit:
FormGenerator JS files are removed, because they are invoked in the
UIscriptImport.ftl. The flag "debug" is inserted.

Commit:
New JS class - Module. It incorportates a title and a content container.

Commit:
InstanceSelector were completely redesigned. It does not call Table
objects any more, but an InstanceBrowser object that displays
the possible existing instances, grouped according to the
descriptor file.

Commit:
The InstanceBrowser contains a Navigator that allows the user to
jump between the different levels of the existing instance selection.

Commit:
The SelectorTable is called as well by the InstanceBrowser and
calls NavigationItem or DataItem objects.

Commit:
New JS UI library functions

Commit:
TextButton were extended to be able set and reset the hower functionality.
Moreover it can return parameters coming as input.

Commit:
TextField style can be set now.

Commit:
ExistingInstanceSelector now calls InstanceSelector object
which initiates the floating window on the entry form.

Commit:
DataController handles the AJAX calls more elegantly. It
can check we are in debug mode, which means the interface uses
a descriptor data defined in the JSONDummy.js.

Commit:
FormLoader now uses the new AJAX call routine

Commit:
Additional JS modifications.

Commit:
New CSS files and style classes

Commit:
Corrections in form descriptor generation.

Commit:
If a FormGraph were defined to a variable on the graph,
then it influences the SPARQLDataGetter for the corresponding
VariableDependency object. The initDataGetter method merges the
restriction triples with the FormGraph triples and creates the
dataGetter field.

Commit:
FormGraph were completely redesigned. It does not support right now
subgraph processing. On the other hand it can handle its Navigator
object, that contains in itself the definition for the multi level
existing instance selection. Navigator can perform the grouping of
the fromgraph data and of course return its descriptor.

Commit:
The field inputNode were renamed to varName.

Commit:
Small Java modifications

Commit:
Debug routine update

Commit:
Identation was changed in the whole file

Commit:
New classes for representing data and form configurations.

Commit:
Default version of FormGraph for easier initialisation

Commit:
Restoring the old state of the triple generator module in Graph.java.
Now the data input works again properly.

Commit:
QueryInfo contains the uris literals and triples for a query.
It is used by the FormGraph class.

Commit:
Older state is restored like in Graph.java. It caused bug in the
data input.

Commit:
Code formatting changed identation.

Commit:
Table.java has new constructor. PlainJavaWebappConnector is added as
well because it is contained in the GraphLib.java.

Commit:
New .n3 and .rdf configuration files for CRIDs class group and
menu item, and for faux properties which connects CRIDs to skeletal
inventories and CRID registries.
classGroupDefinitions.n3 were renamed to classGroupMembership.n3
customDisplay.n3 were renamed to customProfilePage.n3

kdaveed commented:
@zarquon42b 
Please check this branch with the new commits. The most important new feature that it is now possible to add CRIDs and CRID registries, and connect CRIDs to skeletal inventories and CRID registries with faux properties. Moreover the selection of bones for study design execution works only if the bone belongs to a skeletal inventory, which belongs to a CRID, which belongs to a CRID registry.  
So the process to try out the new features : 
* Create an instance of CRID Registry, CRID, Dry Bone Skeletal Inventory, and FrSexEst Investigation
* Go to the profile page of previously create CRID, and on the overview tab you find two faux properties. One for skeletal inventory, one for crid registry. You have to connect the previously created instances with these faux properties.  
* Go to the profile page of the added skeletal inventory and add a complete skull (with the add all buttons)
* Go to profile page of the added investigation and add study design execution
* Then the bone segment button initiates a window that displays the bone grouped by CRID registry, CRID and skeletal inventory
* To select the bone segment you have to click on it

Unfortunately the viewing the existing instances does not work always properly.

zarquon42b commented:
Still cannot edit Skeletal Inventories. Looks like this:
![rdfbones - mozilla firefox_132](https://cloud.githubusercontent.com/assets/1515863/21846071/9ac13090-d7f5-11e6-9194-10a1dc734df6.png)

zarquon42b commented:
> Go to the profile page of previously create CRID, and on the overview tab you find two faux properties. One for skeletal inventory, one for crid registry. You have to connect the previously created instances with these faux properties.

There are no corresponding faux properties, either.

kdaveed commented:
You have to add Dry Bone Skeletal Inventory, not Skeletal Inventory to be able to see the faux properties.

Commit:
Now the JS can handle if an AJAX request fails and it sends
again. If the second fails as well the user gets an alert.

zarquon42b commented:
what are the names of the faux properties?

zarquon42b commented:
OK. I had to wipe everything (including tdbModels) to get it to work

zarquon42b commented on rdf/display/firsttime/studyDesingExecutionFauxProperty.n3:
```diff
@@ -0,0 +1,31 @@
+@prefix obo: <http://purl.obolibrary.org/obo/> .
+
+
+local:studyDesignExecutionContext a :ConfigContext ;
```
<strike>study design execution faux property is already defined in FauxPropertyTest.n3</strike>

zarquon42b replied:
Sorry, forget it. error on my part

Commit:
Graph.java contains the deletor function. It invokes N3 lib which
converts the list of triples and JSON object to string of the
triples to delete.

Commit:
Additional css styling

Commit:
GenericForm calls FormLoader object, not the MainForm.

Commit:
FormLoader is now one individual class and the forms are defined in the
Form.js.

Commit:
DataController call Global object. Identitation corrections.
Bug fixes.

Commit:
New util functions

Commit:
Form is invoked by the scriptImport.ftl

Commit:
The FormElements class is updated to the new type of forms.

Commit:
Due the CSS fade class PopUpController got easier. Now the loader
gifs are extended with a small msg by the form and subform loading.

Commit:
FormMap setter is added. It is used by the FormConfiguration which
performs some postprocessing on the Graph and Form definition to
create the the formNodes and nodesAsInput for the particular graphs.

Commit:
New lib methods

Commit:
Existing data loading is moved to the AJAX controller.

Commit:
Identation change. Delete routine is added.

Commit:
Small changes

Commit:
Selector formelement calls the refreshed DataSetterSelectorFieldMap.
The SelectorField.js is modified to use OOP better.

Commit:
Removed unused method

Commit:
New Log class

Commit:
By the Graph processing the nodeMap of the Graph is created.

Commit:
Now the addition and removal of triples is moved to the WebappConnector.

Commit:
FormGenerator is not needed any more

Commit:
utils contains the redirect function. Now the form submission works
as well through the AJAX library.

Commit:
WebappConnector bug fix.

Commit:
AJAXController handles the data input.

Commit:
DataGenerator is not needed any more. Data addition happens
in AJAXController.

Commit:
N3Lib were renamed to N3

Commit:
N3 import update

Commit:
ReturnFunction needed to be bind with this.

Commit:
Parameter key map in AJAX for data submission

Commit:
Duplicated file

Commit:
New util function

Commit:
Direct function to get if the type of the variable is simple data.

Commit:
Now the Selector form element sends an AJAX request to change
the selected variable if the form is in editing mode.

Commit:
Now the JS framework add deletor buttons to the subform. The
deleteData routine removes the corresponding data object from
the form data.

Commit:
buttonId generation is cleared

Commit:
util function updated

Commit:
Deleting subform on edit is working

Commit:
Removed comments

Commit:
Deleting form data from the form is working.

Commit:
Deleting whole graphData is working

Commit:
InstanceSelector is now rewritten to more elegant and compact class
definition form. And the subclass EditInstanceSelector is added.

Commit:
InstanceBrowser were updated as well to the new class definition.

Commit:
SelectorTable generates the tables so that it cares about the
already added instances.

Commit:
FormElements calls the InstanceSelector or EditInstanceSelector class
depending on if we are on the editing mode.

Commit:
Module class definition is rewritten to ES6 style.
Moreover the UI structure has been simplified and
the 'addObject' routine has been added.

Commit:
New util function that filters the objects of the array by a key
and an input array.

Commit:
InstanceSelector now can display the existing data correctly by getting
out the leaf nodes by the tableData.

Commit:
SelectorTable is updated to the new class definition as well.
Moreover it handles properly if an instance have already been added.

Commit:
InstanceSelector handles the dataItemCache now properly.

Commit:
New JS util function and small modifications.

Commit:
Item is in the new class definition style

Commit:
The AJAX controller files handling the instance deleting and creating
by existing instance selectors.

Commit:
By the existing data loading Graph.java checks if the varName of the subgraph
is not connected to an existing instance selector, becasue if it is the case
then the formData is not loaded as well.

Commit:
New small java modifications

Commit:
FormElements.js is rewritten to new style

Commit:
JSON now can handle boolean types as well by the string generation.

Commit:
New handler for adding subformData.

Commit:
In order respond the graphData JSON object the Graph.java had to be
extended with thes saveDataAJAX(..) routine.

Commit:
New converter routine for Maps

Commit:
Form is rewritten to new style. Moreover it can handle the deletion and
addition of subforms on the edit form properly.

Commit:
FormElements.js is rewritten to support AJAX based data creation
and deletion.

Commit:
New AJAX paramater map

Commit:
Bug fix by the SelectorTable

Commit:
The data created through AJAX is added to the formData properly by
the jQuery $.extend

Commit:
New util function for creating ArrayList from n strings.

Commit:
JSON lib function for generating error messages for AJAX calls.

Commit:
AjaxController checks if all required parameters have arrived from
the client.

Commit:
PopUpController allows to define custom loader messages for
subWaitGif.

Commit:
Selector class bug fixed. It sends the proper AJAX call to change its
value persistently.

Commit:
Now EditSubForm sets each of its form element's flag edit to true
so that multi level data can be added by the edit mode.

Commit:
Deleted unnecessary files.

Commit:
CSS updates

Commit:
CSS import updates

Commit:
Style related updates in Form, FormElements and FormLoader classes

Commit:
Style related updates in UI libs

Commit:
SelectorTable.css is not needed any more.

Commit:
CSS modifications

Commit:
UI function form returning div with click event.

Commit:
resetColor method removes style attribute

Commit:
New class "TableTitle" and the laoder mechanisms are rewritten
a bit.

Commit:
Removed semicolon by FormLoader.js
Niw the SelectorTable adds the table title.
Navigator calls the resetColor() method of the TextButton object.

Commit:
The hover feature does not belong to the generalButton class but to the
hoverClass. TextButton class is rewritten to the ES6 standard, and HomeButton
subclass is added.

Commit:
New small home image. General img src-s are rearranged and home is
inserted.

Commit:
UI has three new functions to return buttons with click event.

Commit:
Navigator now calls the HomeButton object at the initialisation

Commit:
Removed comments

Commit:
Bug fixed. Predicate was rdf:type instead of rdfs:label.

Commit:
Two new fields in Graph.java - labelClasses, labelTriples for automated
triple generation for the new instances. It will have to be reconsidered
for the cases where the form itself contains a label literal field.
GraphLib calculates these vars.

Commit:
getLabels function is not needed any more.

Commit:
GraphLib.getLabelMap calculates the labels for the new instances
based on the labelClasses list. It calls the LabelDataGetter instance
to get the labels of the classes.

Commit:
Small correction

Commit:
Graph class calls by the dataGeneration the new getLabelMap method.

Commit:
New util function to get class label from the uri if the label is not
defined.

Commit:
Correction in the bone organ navigator definition.

Commit:
Removed comments and unnecessary lines.

kdaveed commented:
@zarquon42b The investigationEdit branch were merged into the investigationClean branch. Now the label generation works properly. I think we can merge the branch.
